provider/app.js

```diff
@@ -39,12 +41,76 @@ var monitoringInterval = process.env.MONITORING_INTERVAL;
 var filterDDName = '_design/' + constants.FILTERS_DESIGN_DOC;
 var viewDDName = '_design/' + constants.VIEWS_DESIGN_DOC;
 
+// If the environment variables below does not exist, the discover socket is not created.
+var discoverDestinations = process.env.DISCOVER_DESTINATIONS;
+var discoverAddress = process.env.DISCOVER_ADDRESS || '0.0.0.0';
+var discoverPort = process.env.DISCOVER_PORT || '51007';
+// The following conditions must be met :
+//   masterTimeout >= nodeTimeout >= checkInterval > helloInterval
+var helloInterval = process.env.DISCOVER_HELLO_INTERVAL || 1000;
+var checkInterval = process.env.DISCOVER_CHECK_INTERVAL || 2000;
+var nodeTimeout =  process.env.DISCOVER_NODE_TIMEOUT || 2000;
+var masterTimeout = process.env.DISCOVER_MASTER_TIMEOUT || 2000;
+
 // Create the Provider Server
 var server = http.createServer(app);
 server.listen(app.get('port'), function() {
     logger.info('server.listen', 'Express server listening on port ' + app.get('port'));
 });
 
+
+function createDiscover(destinations) {
+    var method = 'createDiscover';
+
+    return new Promise(function (resolve, reject) {
+        if (destinations) {
+            var opt = {
+                address: discoverAddress,
+                port: discoverPort,
+                unicast: destinations,
```

Comment:
@jasonpet 
It should be the same subnet as it is unitcast.
I run on an in-house platform similar to a docker swarm. So the hosts are the same subnet.

If necessary, I will also add multicast settings. (and broadcast)

Reply:
If your hosts are not on the same subnet, you must bind udp to the port the socket binds to:
```
docker run -d \
   -p 51007:51007/udp \ 
   -e DISCOVER_PORT=51007
```